### PR TITLE
Issue #9419: created example of AST for TokenTypes.DOUBLE_COLON

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Fixes Issue #9419:
Untitled

Java Code Used:
<pre>
<code>
import java.util.function.Function;

public class MyClass {
  public static double square(double num) {
    return Math.pow(num, 2);
  }

Function&lt;Double, Double&gt; square = MyClass::square;
  double ans = square.apply(23d);
}
</code>
</pre>

AST  FORMAT OF Java code:
<pre>
<code>
COMPILATION_UNIT -> COMPILATION_UNIT [1:0]
COMPILATION_UNIT -> COMPILATION_UNIT [1:0]
|--IMPORT -> import [1:0]
|   |--DOT -> . [1:25]
|   |   |--DOT -> . [1:16]
|   |   |   |--DOT -> . [1:11]
|   |   |   |   |--IDENT -> java [1:7]
|   |   |   |   `--IDENT -> util [1:12]
|   |   |   `--IDENT -> function [1:17]
|   |   `--IDENT -> Function [1:26]
|   `--SEMI -> ; [1:34]
`--CLASS_DEF -> CLASS_DEF [3:0]
    |--MODIFIERS -> MODIFIERS [3:0]
    |   `--LITERAL_PUBLIC -> public [3:0]
    |--LITERAL_CLASS -> class [3:7]
    |--IDENT -> MyClass [3:13]
    `--OBJBLOCK -> OBJBLOCK [3:21]
        |--LCURLY -> { [3:21]
        |--METHOD_DEF -> METHOD_DEF [4:2]
        |   |--MODIFIERS -> MODIFIERS [4:2]
        |   |   |--LITERAL_PUBLIC -> public [4:2]
        |   |   `--LITERAL_STATIC -> static [4:9]
        |   |--TYPE -> TYPE [4:16]
        |   |   `--LITERAL_DOUBLE -> double [4:16]
        |   |--IDENT -> square [4:23]
        |   |--LPAREN -> ( [4:29]
        |   |--PARAMETERS -> PARAMETERS [4:30]
        |   |   `--PARAMETER_DEF -> PARAMETER_DEF [4:30]
        |   |       |--MODIFIERS -> MODIFIERS [4:30]
        |   |       |--TYPE -> TYPE [4:30]
        |   |       |   `--LITERAL_DOUBLE -> double [4:30]
        |   |       `--IDENT -> num [4:37]
        |   |--RPAREN -> ) [4:40]
        |   `--SLIST -> { [4:42]
        |       |--LITERAL_RETURN -> return [5:4]
        |       |   |--EXPR -> EXPR [5:19]
        |       |   |   `--METHOD_CALL -> ( [5:19]
        |       |   |       |--DOT -> . [5:15]
        |       |   |       |   |--IDENT -> Math [5:11]
        |       |   |       |   `--IDENT -> pow [5:16]
        |       |   |       |--ELIST -> ELIST [5:20]
        |       |   |       |   |--EXPR -> EXPR [5:20]
        |       |   |       |   |   `--IDENT -> num [5:20]
        |       |   |       |   |--COMMA -> , [5:23]
        |       |   |       |   `--EXPR -> EXPR [5:25]
        |       |   |       |       `--NUM_INT -> 2 [5:25]
        |       |   |       `--RPAREN -> ) [5:26]
        |       |   `--SEMI -> ; [5:27]
        |       `--RCURLY -> } [6:2]
        |--VARIABLE_DEF -> VARIABLE_DEF [8:2]
        |   |--MODIFIERS -> MODIFIERS [8:2]
        |   |--TYPE -> TYPE [8:2]
        |   |   |--IDENT -> Function [8:2]
        |   |   `--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [8:10]
        |   |       |--GENERIC_START -> < [8:10]
        |   |       |--TYPE_ARGUMENT -> TYPE_ARGUMENT [8:11]
        |   |       |   `--IDENT -> Double [8:11]
        |   |       |--COMMA -> , [8:17]
        |   |       |--TYPE_ARGUMENT -> TYPE_ARGUMENT [8:19]
        |   |       |   `--IDENT -> Double [8:19]
        |   |       `--GENERIC_END -> > [8:25]
        |   |--IDENT -> square [8:27]
        |   |--ASSIGN -> = [8:34]
        |   |   `--EXPR -> EXPR [8:43]
        |   |       `--METHOD_REF -> :: [8:43]
        |   |           |--IDENT -> MyClass [8:36]
        |   |           `--IDENT -> square [8:45]
        |   `--SEMI -> ; [8:51]
        |--VARIABLE_DEF -> VARIABLE_DEF [9:2]
        |   |--MODIFIERS -> MODIFIERS [9:2]
        |   |--TYPE -> TYPE [9:2]
        |   |   `--LITERAL_DOUBLE -> double [9:2]
        |   |--IDENT -> ans [9:9]
        |   |--ASSIGN -> = [9:13]
        |   |   `--EXPR -> EXPR [9:27]
        |   |       `--METHOD_CALL -> ( [9:27]
        |   |           |--DOT -> . [9:21]
        |   |           |   |--IDENT -> square [9:15]
        |   |           |   `--IDENT -> apply [9:22]
        |   |           |--ELIST -> ELIST [9:28]
        |   |           |   `--EXPR -> EXPR [9:28]
        |   |           |       `--NUM_DOUBLE -> 23d [9:28]
        |   |           `--RPAREN -> ) [9:31]
        |   `--SEMI -> ; [9:32]
        `--RCURLY -> } [10:0]

</code>
</pre>

Expected Update for Java Doc
<pre>
<code>
        |--VARIABLE_DEF -> VARIABLE_DEF [8:2]
        |   |--MODIFIERS -> MODIFIERS [8:2]
        |   |--TYPE -> TYPE [8:2]
        |   |   |--IDENT -> Function [8:2]
        |   |   `--TYPE_ARGUMENTS -> TYPE_ARGUMENTS [8:10]
        |   |       |--GENERIC_START -> < [8:10]
        |   |       |--TYPE_ARGUMENT -> TYPE_ARGUMENT [8:11]
        |   |       |   `--IDENT -> Double [8:11]
        |   |       |--COMMA -> , [8:17]
        |   |       |--TYPE_ARGUMENT -> TYPE_ARGUMENT [8:19]
        |   |       |   `--IDENT -> Double [8:19]
        |   |       `--GENERIC_END -> > [8:25]
        |   |--IDENT -> square [8:27]
        |   |--ASSIGN -> = [8:34]
        |   |   `--EXPR -> EXPR [8:43]
        |   |       `--METHOD_REF -> :: [8:43]
        |   |           |--IDENT -> MyClass [8:36]
        |   |           `--IDENT -> square [8:45]
        |   `--SEMI -> ; [8:51]
</code>
</pre>

MVN CLEAN VERIFY
MVN

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
@romani @rnveach please review my changes as it is the last day of hacktoberfest.

**Comment by Rahulkhinchi03** [comment]
@romani sir, please review my changes!

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -2341,14 +2341,40 @@
      * @see #CASE_GROUP
      **/
     public static final int COLON = JavaLanguageLexer.COLON;
-
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2344]: Restore separation

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -2341,14 +2341,40 @@
...
+     *  |   |   `--EXPR -&gt; EXPR
+     *  |   |       `--METHOD_REF -&gt; ::
+     *  |   |           |--IDENT -&gt; MyClass
+     *  |   |           `--IDENT -&gt; square
+     *  |   `--SEMI -&gt; ;
+     * </pre>
+     *
      * @see #METHOD_REF
-     */
+     **/
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:2377]: Restore

**Review by romani — CHANGES_REQUESTED** [review]
Items:

**Review by romani — APPROVED** [review]

**Pull request merged**